### PR TITLE
added React Todo App

### PR DESCRIPTION
**Pull request by nirban256**
# Description

I have added a React Todo List App which tracks the task that is being specified with the functionality of adding the todos in the  Local Storage

## Type of change

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

# Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

**Comment by nirban256** [comment]
Please review the pull request @atapas

**Comment by atapas** [comment]
> Please review the pull request @atapas

@nirban256 great, we will start the review. Could you please create an issue for this play and associate it to the PR

File: src/plays/react-todo-app/react-todo.css [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+    font-family: "Poppins", sans-serif;
+    min-height: 100vh;
+}
+
+.top {
+    font-size: 2rem;
+}
+
+.top,
+form {
```
Comment (atapas) [inline src/plays/react-todo-app/react-todo.css:19]: don't use the HTML selectors directly in CSS as they may conflict with an existing style. Please scope them with a class name related to your play.

File: src/plays/react-todo-app/react-todo.css [hunk 2]
```diff
@@ -0,0 +1,152 @@
...
+    opacity: 0;
+}
+
+.completed {
+    text-decoration: line-through;
+    opacity: 0.5;
+}
+
+/*CUSTOM SELECTOR */
+select {
```
Comment (atapas) [inline src/plays/react-todo-app/react-todo.css:110]: Same comment as form

File: src/plays/react-todo-app/react-todo.css [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+
+/* Custom Select */
+.select {
+    margin: 1rem;
+    position: relative;
+    overflow: hidden;
+}
+
+select {
+    color: #ff6f47;
```
Comment (atapas) [inline src/plays/react-todo-app/react-todo.css:129]: Above we have style for select, why repeating here?

File: src/plays/react-todo-app/react-todo.css [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+    padding: 1rem;
+    background: #ff6f47;
+    cursor: pointer;
+    pointer-events: none;
+}
+
+/* Transition */
+/*
+.select:hover::after {
+    transform: scale(1.5);
```
Comment (atapas) [inline src/plays/react-todo-app/react-todo.css:150]: remove if not required.

File: src/plays/react-todo-app/components/Input.jsx [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+const Input = ({ input, setInput, todos, setTodos, setStatus }) => {
+
+    const inputHandler = (e) => {
+        setInput(e.target.value);
+    }
+
+    const submitHandler = (e) => {
+        e.preventDefault();
+        setTodos([
```
Comment (atapas) [inline src/plays/react-todo-app/components/Input.jsx:12]: Don't allow to add an empty TODO

File: src/plays/react-todo-app/components/TodoList.jsx [hunk 6]
```diff
@@ -0,0 +1,16 @@
+import React from 'react'
+import Todo from './Todo';
+
+const TodoList = ({ todos, setTodos, filtered }) => {
+    return (
+        <div className="todo-container">
+            <ul className="todo-list">
+                {filtered.map((todo) => (
```
Comment (atapas) [inline src/plays/react-todo-app/components/TodoList.jsx:8]: We can show a message if the filtered is empty... like no TODO found matching the criteria.

File: src/plays/react-todo-app/ReactTodoApp.jsx [hunk 7]
```diff
@@ -0,0 +1,84 @@
...
+          setFiltered(todos.filter(todo => todo.completed === false));
+          break;
+        default:
+          setFiltered(todos);
+          break;
+      }
+    }
+
+    const saveLocalTodos = () => {
+      localStorage.setItem("todos", JSON.stringify(todos));
```
Comment (atapas) [inline src/plays/react-todo-app/ReactTodoApp.jsx:59]: Change the key name by prefixing your github id. Soemone else may add a todo in future with will override it.

**Review by atapas — CHANGES_REQUESTED** [review]
@nirban256 A few comments

**Comment by atapas** [comment]
Hello @nirban256,

We have upgraded to React v18 as part of this issue: https://github.com/atapas/react-play/issues/87

We have tested the existing plays. All good. Please pull this change to your branch and keep it up-to-date. You have to do sanity of things to make sure everything works for your changes in your branch.

Happy Coding!

**Comment by nirban256** [comment]
Hello @atapas Sir can I do another pull request by doing all the changes by associating it with an issue raised by me?

**Comment by atapas** [comment]
> Hello @atapas Sir can I do another pull request by doing all the changes by associating it with an issue raised by me?

You can, just discard this one in that case.

**Pull request closed without merging**